lineapy/api/models/pipeline.py

```diff
@@ -0,0 +1,187 @@
+        session_orm = (
+            db.session.query(SessionContextORM)
+            .order_by(SessionContextORM.creation_time.desc())
+            .all()
+        )
+        if len(session_orm) == 0:
+            track(ExceptionEvent(ErrorType.PIPELINE, "No session found in DB"))
+            raise Exception("No sessions found in the database.")
+        last_session = session_orm[0]
+
+        if framework in PipelineType.__members__:
+            if PipelineType[framework] == PipelineType.AIRFLOW:
+
+                ret = AirflowPlugin(db, last_session.id).sliced_airflow_dag(
+                    self.name,
+                    self.task_graph,
+                    output_dir=output_dir,
+                    airflow_dag_config=pipeline_dag_config,
+                )
+
+            else:
+
+                ret = ScriptPlugin(db, last_session.id).sliced_pipeline_dag(
+                    self.name,
+                    self.task_graph,
+                    output_dir=output_dir,
+                )
+
+            # send the info
+            track(
+                ToPipelineEvent(
+                    framework,
+                    len(self.artifact_names),
+                    self.task_graph.get_airflow_dependency() != "",
+                    pipeline_dag_config is not None,
+                )
+            )
+            return ret
+
+        else:
+            raise Exception(f"No PipelineType for {framework}")
+
+    def export2(
+        self,
+        framework: str = "SCRIPT",
+        output_dir: str = ".",
+        pipeline_dag_config: Optional[AirflowDagConfig2] = {},
+    ) -> None:
+        """
+        This method uses objects implementing new style of graph refactor.
+        It is meant to eventually replace existing `export()` method above.
+        TODO: Replace existing `export()` method with this one.
+        """
+        # Create artifact collection
+        execution_context = get_context()
+        artifact_collection = ArtifactCollection(
+            execution_context.executor.db, self.artifact_names
+        )
+
+        # Check if the specified framework is a supported/valid one
+        if framework not in PipelineType.__members__:
+            raise Exception(f"No PipelineType for {framework}")
+
+        # Determine writer class to use
+        pipeline_writer_class = {
+            PipelineType.SCRIPT: BasePipelineWriter,
+            PipelineType.AIRFLOW: AirflowPipelineWriter,
+        }[PipelineType[framework]]
+
+        # Construct pipeline writer
+        pipeline_writer = pipeline_writer_class(
+            artifact_collection=artifact_collection,
+            dependencies=self.dependencies,
+            pipeline_name=self.name,
+            output_dir=output_dir,
+            dag_config=pipeline_dag_config,
+        )
+
+        # Write out pipeline files
+        pipeline_writer.write_pipeline_files()
+
+        # # Track the event
+        # # TODO: Uncomment once this replaces old `export()` method
```

Comment:
For this and the previous todo, if you have a ticket open for it, can we make the comment formatted like 

`TODO [LIN-XXX]` so we can track the todos in our codebase?